### PR TITLE
OpenThread RCP mode integration into Zephyr

### PR DESCRIPTION
**Pull request by markus-becker-tridonic-com**
* Add RCP library.
* Conditionally remove non required libraries not required for RCP.

Relates to https://github.com/zephyrproject-rtos/zephyr/issues/26383.

### CODE REVIEW

File: zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -235,29 +235,39 @@ zephyr_system_include_directories(../examples/platforms)
 # Determine which libs should be linked in
 set(ot_libs "")
 
+if(CONFIG_OPENTHREAD_COPROCESSOR_RCP)
```
Comment (rlubos) [inline zephyr/CMakeLists.txt:238]: This does not seem to be needed, the CLI libs will only be linked when shell is enabled (which is blocked for coprocessor now).

File: zephyr/CMakeLists.txt [hunk 2]
```diff
@@ -235,29 +235,39 @@ zephyr_system_include_directories(../examples/platforms)
...
+
+if(CONFIG_OPENTHREAD_COPROCESSOR_NCP)
 if(CONFIG_OPENTHREAD_FTD)
 list(APPEND ot_libs openthread-ncp-ftd)
 elseif(CONFIG_OPENTHREAD_MTD)
 list(APPEND ot_libs openthread-ncp-mtd)
 endif()
 endif()
 
+if(CONFIG_OPENTHREAD_COPROCESSOR_RCP)
```
Comment (rlubos) [inline zephyr/CMakeLists.txt:263]: Let's make it `if (NOT CONFIG_OPENTHREAD_COPROCESSOR_RCP)`

File: zephyr/CMakeLists.txt (hunk 1 above)
Comment (markus-becker-tridonic-com) [reply zephyr/CMakeLists.txt]: Yes. Applied.

File: zephyr/CMakeLists.txt (hunk 2 above)
Comment (markus-becker-tridonic-com) [reply zephyr/CMakeLists.txt]: Applied.

File: zephyr/CMakeLists.txt [hunk 3]
```diff
@@ -249,19 +249,25 @@ if(CONFIG_OPENTHREAD_SHELL)
...
+
+if(CONFIG_OPENTHREAD_COPROCESSOR_NCP)
 if(CONFIG_OPENTHREAD_FTD)
 list(APPEND ot_libs openthread-ncp-ftd)
 elseif(CONFIG_OPENTHREAD_MTD)
 list(APPEND ot_libs openthread-ncp-mtd)
 endif()
 endif()
 
+if(NOT CONFIG_OPENTHREAD_COPROCESSOR_RCP)
```
Comment (tejlmand) [inline zephyr/CMakeLists.txt:264]: Why is this not just a:
```
if(CONFIG_OPENTHREAD_COPROCESSOR_RCP)
  list(APPEND ot_libs openthread-rcp)
else()
  if(CONFIG_OPENTHREAD_FTD)
    list(APPEND ot_libs openthread-ncp-ftd)
  elseif(CONFIG_OPENTHREAD_MTD)
    list(APPEND ot_libs openthread-ncp-mtd)
  endif()
endif()

**Review by tejlmand — CHANGES_REQUESTED** [review]

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
@tejlmand Could you please revisit this PR?

File: zephyr/CMakeLists.txt (hunk 3 above)
Comment (tejlmand) [reply zephyr/CMakeLists.txt:264]: accepted

**Review by tejlmand — APPROVED** [review]

**Pull request merged**